crates/shared/src/price_estimation/oneinch.rs

```diff
@@ -1,15 +1,22 @@
 use crate::{
-    oneinch_api::{OneInchClient, ProtocolCache, RestResponse, SellOrderQuoteQuery},
+    oneinch_api::{
+        OneInchClient, ProtocolCache, RestResponse, SellOrderQuote, SellOrderQuoteQuery,
+    },
     price_estimation::{
         gas, Estimate, PriceEstimateResult, PriceEstimating, PriceEstimationError, Query,
     },
+    request_sharing::RequestSharing,
 };
-use futures::StreamExt;
+use futures::{future::BoxFuture, FutureExt, StreamExt};
 use model::order::OrderKind;
 use std::sync::Arc;
 
 pub struct OneInchPriceEstimator {
     api: Arc<dyn OneInchClient>,
+    sharing: RequestSharing<
+        Query,
+        BoxFuture<'static, Result<RestResponse<SellOrderQuote>, PriceEstimationError>>,
```

Comment:
Slightly awkward here is that the future must be 'static even though a hypothetical 'self would fit better.

Reply:
Self-referential struct 🙅

Reply:
Self-referential struct 🙅